### PR TITLE
Add `all_deps_blocking_checks_passed` condition

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

As title

## Changelog

> Insert changelog entry or delete this section.

### CODE REVIEW

**Comment by OwenKephart** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25171?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25171?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25170?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py [hunk 1]
```diff
@@ -518,6 +518,20 @@ def any_deps_in_progress() -> "AnyDepsCondition":
                 "any_deps_in_progress"
             )
 
+    @experimental
+    @staticmethod
+    def all_deps_blocking_checks_passed() -> "AllDepsCondition":
+        """Returns an AutomationCondition that is true for any partition where all upstream
+        blocking checks have passed, or will be requested.
+        """
```
Comment (schrockn) [inline python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py:526]: Why is "will be requested" included here where the top-level language is "passed"?

The docblock should explain this.

**Review by schrockn — APPROVED** [review]
Please expand on what is going on here in docblock.

**Comment by OwenKephart** [comment]
### Merge activity

* **Oct 15, 2:48 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/25171).

**Pull request merged**